store/sqlstore/license_store.go

```diff
@@ -52,9 +54,17 @@ func (ls SqlLicenseStore) Save(license *model.LicenseRecord) (*model.LicenseReco
 		return nil, errors.Wrap(err, "license_tosql")
 	}
 	var storedLicense model.LicenseRecord
-	if err := ls.GetReplica().SelectOne(&storedLicense, queryString, args...); err != nil {
+	if err := ls.GetReplicaX().Get(&storedLicense, queryString, args...); err != nil {
 		// Only insert if not exists
-		if err := ls.GetMaster().Insert(license); err != nil {
+		insertLicense, insertArgs, err := ls.getQueryBuilder().Insert("Licenses").SetMap(sq.Eq{
+			"Id":       license.Id,
+			"CreateAt": license.CreateAt,
+			"Bytes":    license.Bytes,
+		}).ToSql()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get licenseId=%s", license.Id)
+		}
+		if _, err = ls.GetMasterX().NamedExec(insertLicense, insertArgs); err != nil {
```

Comment:
`NamedExec` does not work with `args...`. You would need to use the old style of `INSERT INTO table (col1, col2) VALUES (:col1, :col2)` and then pass the struct in `NamedExec`

Reply:
Think I'll close the pr.
I like the e2e coverage aspect on this repo. But unit testing is quite nigh impossible with the sql approach.